introduction.md

```diff
@@ -1,98 +1,51 @@
-
+This book is a practical—and we hope, entertaining—introduction to some
+of the most important data structures and algorithms in computer
+science.
+
+We have written the examples in this book in idiomatic Python to help
+convey the *usefulness* of these data structures and algorithms, both
+directly and as models for analogous work that you will do.
+
+Many software engineers are skeptical that this subject matter is
+useful. Our best explanation for such skepticism is that those engineers
+never encountered these topics in a way that was encouraged deep
+understanding. Perhaps they were [unfortunate
+enough](http://v.cx/2010/04/feynman-brazil-education) to participate in
+a computer science program that encouraged rote memorization, or
+subjected themselves to such an experience for the sake of succeeding at
+technical interviews. Or perhaps they became adequate programmers
+without having encountered the content at all, and presume that they
+have reached the limits of their capabilities.
+
+Whatever the case, we’re confident that the usefulness of studying
+algorithms and data structures increases proportionally to the depth at
+which you understand it—a depth that comes through a combination of
+study and application. Those who understand trees in depth are more
+likely to identify tree shapes in their own data, and have the
+confidence to apply a traversal algorithm appropriately. Those who count
+Big-O analysis as a tool in their toolkits not only have a better
+vocabulary for comparing alternative strategies, they are also more
+cognizant overall of the trade offs we implicitly make every day. Those
+who see recursion everywhere see recursion everywhere :)
+
+One trade off that *we* have made is to sacrifice breadth for the sake
+of depth. We cover fewer data structures and algorithms than a typical
+introductory text may cover, in order to focus more on what we consider
+to be important. For instance we do not cover sorting algorithms at all,
+and instead spend more time walking you through interesting applications
+of tree and graph traversal algorithms. For this reason, we hope that
+you work through the examples slowly, with a REPL or text editor open,
+verifying that you absorb and appreciate the content at the depth
+required to appreciate its usefulness for decades to come.
 
 Assumed Knowledge
 ---
 
-This book assumes intermediate knowledge of Python. If you encounter any unfamiliar Python usage, we encourage you to use [Learn Python the Hard Way](http://learnpythonthehardway.org/) or [Dive into Python](http://www.diveintopython.net/toc/index.html), both of which are excellent, free
+This book assumes intermediate knowledge of Python. If you encounter any
+unfamiliar Python usage, we encourage you to use [Learn Python the Hard
+Way](http://learnpythonthehardway.org/) or [Dive into
+Python](http://www.diveintopython.net/toc/index.html), both of which are
+excellent and freely available online.
```

Comment:
Yeah maybe referring to an intro text is not the best advice. Thinking about this some more, it may be best to just link out to specific resources whenever there's a code sample with non-obvious Python usage. So I'm going to remove this section entirely.